src/components/NftPreview/NftPreview.tsx

```diff
@@ -98,11 +104,16 @@ function NftPreview({
           width={nftPreviewWidth}
           backgroundColorOverride={backgroundColorOverride}
         >
-          <NftPreviewAsset
-            tokenRef={token}
-            // we'll request images at double the size of the element so that it looks sharp on retina
-            size={previewSize * 2}
-          />
+          {token.media?.__typename === 'UnknownMedia' ? (
```

Comment:
@Robinnnnn Is there any edge cases that I'm missing by checking the media type? 

Lmk if we would like to change on how to validate failed nft. 

Other thing I can think of is by using `graphqlGetResizedNftImageUrlWithFallback` function

Reply:
@jakzaizzat so we basically need to use this: `getVideoOrImageUrlForNftPreview`, which might be the function you're referring to, to see whether a URL is available or not.

`UnknownMedia` does not necessarily mean we don't have a URL... it simply means the backend failed to categorize what kind of asset it was. BUT it could definitely still have a URL that is render-able!

Reply:
Thanks for the info robin! I misunderstood this `UnknownMedia` type

I've updated the `getVideoOrImageUrlForNftPreview` with additional flag